### PR TITLE
User preference endpoints (#704)

### PR DESCRIPTION
**Pull request by Mefpef**
Please DO NOT DELETE this template. Fill it. PR without your declaration that you licence your code under MIT will 
be discarded. 

Tips
----------------

 * Give the same name to the pull request as the name of ticket if possible
 * Make the title in form of: Pull request title (#<issue-id>)
 * Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 * Test your code thoroughly before PR.

Pull Request title
-------------------

     User preference endpoints (#704)

Pull Request description
-------------------------

   Filled endpoints:
   GET /user/preferences
   DELETE /user/preference/{key}
   POST /user/prefernece/{key}

----

We need your declaration that your contribution is MIT licensed without it, we can not merge your code.
To make clear that you license your contribution under the MIT you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licensed under the [MIT license](https://opensource.org/licenses/MIT)

### CODE REVIEW

File: app/api/user_endpoint.py [hunk 1]
```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
     """
     Sets user preference.Uses key to set the preference
```
Comment (DawidekZagajnik) [inline app/api/user_endpoint.py:37]: no idea what it does from this docstring

File: app/api/user_endpoint.py [hunk 2]
```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
...
     """
-    pass
+    user_preference = storage.driver.user.preference[key]
```
Comment (DawidekZagajnik) [inline app/api/user_endpoint.py:39]: storage.driver.user does not seem to have preference attribute with __getitem__ dunder, you get user from args

File: app/api/user_endpoint.py [hunk 3]
```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
...
+    if user_preference:
+        user_preference.update({key: preference})
```
Comment (DawidekZagajnik) [inline app/api/user_endpoint.py:41]: update user's preferences in User object that you get from args and then save it

File: app/api/user_endpoint.py [hunk 4]
```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
...
+
+    return {"info": "User preference updated"}
```
Comment (DawidekZagajnik) [inline app/api/user_endpoint.py:43]: return driver response after upserting data

File: app/api/user_endpoint.py [hunk 5]
```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
...
 
 
 @router.delete("/user/preference/{key}", tags=["user"], include_in_schema=server.expose_gui_api, response_model=dict)
 async def delete_user_preference(key: str, user=Depends(Permissions(["admin", "developer", "marketer", "maintainer"]))):
     """
     Deletes user preference
     """
-    pass
+    del user.preference[key]
```
Comment (DawidekZagajnik) [inline app/api/user_endpoint.py:51]: great, now save user using driver

File: app/api/user_endpoint.py [hunk 6]
```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
...
+
+    return {"Info": "Preference deleted"}
```
Comment (DawidekZagajnik) [inline app/api/user_endpoint.py:53]: and return driver response

File: app/api/user_endpoint.py [hunk 7]
```diff
@@ -36,23 +36,29 @@ async def set_user_preference(key: str, preference: Union[dict, str, int, float]
...
 
 
 @router.get("/user/preferences", tags=["user"], include_in_schema=server.expose_gui_api, response_model=dict)
 async def gets_all_user_preferences(user=Depends(Permissions(["admin", "developer", "marketer", "maintainer"]))):
     """
     Returns all user preferences
     """
-    pass
+    return user.preference
```
Comment (DawidekZagajnik) [inline app/api/user_endpoint.py:61]: seems to work OK

**Review by DawidekZagajnik — CHANGES_REQUESTED** [review]

**Review by atompie — APPROVED** [review]

**Pull request merged**